### PR TITLE
Bug fix in Div E computation for RZ PSATD

### PR DESCRIPTION
**Pull request by RemiLehe**
**These code changes were extracted from #1216. Many thanks to @EZoni and @dpgrote for spotting and fixing this bug.**

In the RZ PSATD mode, all the fields are computed on the cell centers. However, div(E) was still computed on the nodes, which lead to a discrepancy. This PR fixes the issue.

### CODE REVIEW

**Review by EZoni — APPROVED** [review]
Thanks for splitting #1216 in separate PRs!

File: Source/Diagnostics/ComputeDiagFunctors/DivEFunctor.cpp [hunk 1]
```diff
@@ -20,11 +20,16 @@ DivEFunctor::operator()(amrex::MultiFab& mf_dst, const int dcomp, const int /*i_
     // output Multifab, mf_dst, the guard-cell data is not needed especially considering
     // the operations performend in the CoarsenAndInterpolate function.
     constexpr int ng = 1;
+#if (defined WARPX_DIM_RZ) && (defined WARPX_USE_PSATD)
+    // For RZ spectral, all quantities are cell centered.
+    amrex::IntVect typ = amrex::IntVect::TheCellVector();
```
Comment (ax3l) [inline Source/Diagnostics/ComputeDiagFunctors/DivEFunctor.cpp]:
```suggestion
    amrex::IntVect cell_type = amrex::IntVect::TheCellVector();
```

File: Source/Diagnostics/ComputeDiagFunctors/DivEFunctor.cpp [hunk 2]
```diff
@@ -20,11 +20,16 @@ DivEFunctor::operator()(amrex::MultiFab& mf_dst, const int dcomp, const int /*i_
...
+#else
     // For staggered and nodal calculations, divE is computed on the nodes.
     // The temporary divE MultiFab is generated to comply with the location of divE.
-    const amrex::BoxArray& ba = amrex::convert(warpx.boxArray(m_lev),
-                                amrex::IntVect::TheUnitVector());
-    amrex::MultiFab divE(ba, warpx.DistributionMap(m_lev), 2*warpx.n_rz_azimuthal_modes-1, ng);
+    amrex::IntVect typ = amrex::IntVect::TheNodeVector();
```
Comment (ax3l) [inline Source/Diagnostics/ComputeDiagFunctors/DivEFunctor.cpp]:
```suggestion
    amrex::IntVect cell_type = amrex::IntVect::TheNodeVector();
```

File: Source/Diagnostics/ComputeDiagFunctors/DivEFunctor.cpp [hunk 3]
```diff
@@ -20,11 +20,16 @@ DivEFunctor::operator()(amrex::MultiFab& mf_dst, const int dcomp, const int /*i_
...
+#endif
+    const amrex::BoxArray& ba = amrex::convert(warpx.boxArray(m_lev),typ);
```
Comment (ax3l) [inline Source/Diagnostics/ComputeDiagFunctors/DivEFunctor.cpp]:
```suggestion
    const amrex::BoxArray& ba = amrex::convert(warpx.boxArray(m_lev), cell_type);
```

**Review by ax3l — APPROVED** [review]
Well done, y'all! :sparkles: 

Added a minor spelling suggestion :)

**Pull request merged**